extensions/2.0/Vendor/EXT_texture_webp/README.md

```diff
@@ -0,0 +1,110 @@
+# EXT_texture_webp
+
+## Contributors
+
+* Omar Shehata, Cesium, [@OmarShehata](https://github.com/OmarShehata)
+* Pascal Massimino, Google, [@Skal65535](https://github.com/skal65535)
+
+## Status
+
+Draft
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+This extension allows glTF models to use WebP as a valid image format. A client that does not implement this extension can ignore the provided WebP image and continue to rely on the PNG and JPG textures available in the base specification. Defining a fallback texture is optional. The [best practices](#best-practices) section describes the intended use case of this extension and the expected behavior when using it without a fallback texture.
+
+WebP is an image format developed and maintained by Google that provides superior lossless and lossy compression rates for images on the web. It is typically 26% smaller in size compared to PNGs and 25-34% smaller than comparable JPEG images - [source](https://developers.google.com/speed/webp/).  
+
+## glTF Schema Updates
+
+The `EXT_texture_webp` extension is added to the `textures` node and specifies a `source` property that points to the index of the `images` node which in turn points to the WebP image.
+
+The following glTF will load `image.webp` in clients that support this extension, and otherwise fall back to `image.png`.
+
+```
+"textures": [
+    {
+        "source": 0,
+        "extensions": {
+            "EXT_texture_webp": {
+                "source": 1
+            }
+        }
+    }
+],
+"images": [
+    {
+        "uri": "image.png"
+    },
+    {
+        "uri": "image.webp"
+    }
+]
+```
+
+When used in the glTF Binary (.glb) format the `images` node that points to the WebP image uses the `mimeType` value of `image/webp`. 
+
+```
+"textures": [
+    {
+        "source": 0,
+        "extensions": {
+            "EXT_texture_webp": {
+                "source": 1
+            }
+        }
+    }
+],
+"images": [
+    {
+        "mimeType": "image/png",
+        "bufferView": 1
+    },
+    {
+        "mimeType": "image/webp",
+        "bufferView": 2
+    }
+]
+```
+
+### JSON Schema
+
+[glTF.EXT_texture_webp.schema.json](schema/glTF.EXT_texture_webp.schema.json)
+
+## Best Practices
```

Comment:
Can we provide suggestions regarding best-suitable WebP compression modes for each glTF texture type (ORM, normal map, etc)?

Reply:
@lexaknyazev It would be handy to have such info, but I'm not sure that's in scope here.  Is there something you had in mind?

Reply:
`cwebp` tool has a long list of command-line options. These could be of our interest:
```
  -q <float> ............. quality factor (0:small..100:big), default=75
  -alpha_q <int> ......... transparency-compression quality (0..100),
                           default=100
  -preset <string> ....... preset setting, one of:
                            default, photo, picture,
                            drawing, icon, text
     -preset must come first, as it overwrites other parameters
  -z <int> ............... activates lossless preset with given
                           level in [0:fast, ..., 9:slowest]
  -exact ................. preserve RGB values in transparent area, default=off
  -noalpha ............... discard any transparency information
  -lossless .............. encode image losslessly, default=off
  -near_lossless <int> ... use near-lossless image
                           preprocessing (0..100=off), default=100
```
JPEG-like artifacts are noticeably bad for normal maps. So having a hint (my guess is `-noalpha` and `-lossless`) in the spec regarding this use case should benefit users of this extension.

Reply:
At Cesium we've so far been using WebP for photogrammetry models which have all the lighting baked in so I don't know that I can speak with confidence about best practices for the general case. I did add a note about lossless compression for normal maps and I found this [WebP Compression Study](https://developers.google.com/speed/webp/docs/webp_study) a useful reading so I added it to the resources at the bottom as well.

Reply:
I suggested stripping alpha just in case original file (most likely PNG) was saved in RGBA mode. It has nothing to do with artifacts. We can proceed with this extension as it is and update recommended compression settings later based on production experience.

Reply:
That sounds good with me. I removed that note but left the compression study in the resources.